src/react/actions/s3object.js

```diff
@@ -533,3 +543,23 @@ export function putObjectTagging(
       .finally(() => dispatch(networkEnd()));
   };
 }
+
+export function putObjectLegalHold(
+  bucketName: string,
+  objectKey: string,
+  versionId: string,
+  isLegalHold: boolean,
+  prefixWithSlash: string,
+): ThunkStatePromisedAction {
+  return (dispatch, getState) => {
+    const { zenkoClient } = getClients(getState());
+    dispatch(networkStart('Getting object legal hold'));
+    return zenkoClient
+      .putObjectLegalHold(bucketName, objectKey, versionId, isLegalHold)
+      .then(() => dispatch(getObjectMetadata(bucketName, objectKey, versionId)))
+      .then(() => dispatch(listObjects(bucketName, prefixWithSlash)))
```

Comment:
Maybe we could just dispatch the action that select the object here temporarily while the url rework is not done ?

Reply:
that's right!!! I will just reverse the L.559 and L.560